Android build failing

The failure is due to our `CoreThreadLinux` adjustments in #786.  Here's the error:

```
src/autowiring/CoreThreadLinux.cpp: In member function 'void BasicThread::SetThreadPriority(ThreadPriority)':
src/autowiring/CoreThreadLinux.cpp:37:14: error: 'SCHED_IDLE' was not declared in this scope
     policy = SCHED_IDLE;
```

We need to fix this, or turn off this feature on Android.

--- a/src/autowiring/CoreThreadLinux.cpp
+++ b/src/autowiring/CoreThreadLinux.cpp
@@ -34,7 +34,10 @@ void BasicThread::SetThreadPriority(ThreadPriority threadPriority) {
 
   switch (threadPriority) {
   case ThreadPriority::Idle:
+//Android kernel(3.10.x) has not implemented SCHED_IDLE yet.
+#ifndef __ANDROID__
     policy = SCHED_IDLE;
+#endif
     percent = 0;
     break;
   case ThreadPriority::Lowest:

